From 4a3930eb43c3c50735f98394d8c5be248681d4bb Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Tue, 22 Feb 2022 18:36:45 +0200
Subject: [PATCH] [SW-65638] gaudi2: release: change oam reset args

oam-reset on and off are not operational, the only one that works it's
the cycle, which is the same as the cpld_reset procedure for the SA.
now --oam-reset will invoke a reset cycle.

Change-Id: I41d14a945e16a6bda6ccfe0e1e1c457b101b6c05
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/hls_platform.rc b/scripts/fwi/hls_platform.rc
index 8cc97af..503b0d3 100644
--- a/scripts/fwi/hls_platform.rc
+++ b/scripts/fwi/hls_platform.rc
@@ -29,8 +29,7 @@
 {
 cat << END>&3
 	--open-uart                                  : open the uart on the bmc-oam server, default is to open
-	--oam-reset cycle                            : controls the OAM reset - for now only cycle
-	                                               is supported. (SA cpld_power_cycle)
+	--oam-reset                                  : a CPLD OAM reset (SA cpld_power_cycle)
 	--oam-full-pwr <on|off|cycle>                : controls the OAM power (SA full power cycle)
 	--oam-board-pwr <on|off|cycle>               : controls the OAM power (SA board power cycle)
 	--hib-power <on|off|cycle>                   : power cycle the HiB via the bmc-oam server
@@ -939,7 +938,6 @@
 {
 
 	[ ${oam_reset_f:-0} -eq 0 ] && return 0
-	[ ${#oam_reset_args[@]} -lt 1 ] && return 1
 
 	case ${oam_reset_args[0]} in
 	0|off)
@@ -952,6 +950,7 @@
 		run_per_selected_devs oam_reset_cycle
 		;;
 	NONE)
+		run_per_selected_devs oam_reset_cycle
 		;;
 	*)
 		pwarn "[HLS] uknown reset mode argument: ${oam_reset_args[0]}. Ignoring"
